codeinputlib/src/main/java/com/github/glomadrian/codeinputlib/CodeInput.java

```diff
@@ -185,14 +185,14 @@ protected void onFocusChanged(boolean gainFocus, int direction, Rect previouslyF
   }
 
   @Override protected void onSizeChanged(int w, int h, int oldw, int oldh) {
-    super.onSizeChanged(w, (int) viewHeight, oldw, oldh);
+    super.onSizeChanged((int) ((underlineWidth + underlineReduction) * DEFAULT_CODES), (int) viewHeight, oldw, oldh);
     height = h;
     initUnderline();
   }
 
   @Override protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
     super.onMeasure(widthMeasureSpec, heightMeasureSpec);
-    setMeasuredDimension(getMeasuredWidth(), (int) viewHeight);
+    setMeasuredDimension((int) ((underlineWidth + underlineReduction) * DEFAULT_CODES), (int) viewHeight);
```

Comment:
I don't think `underlineReduction` is needed here, as it doesn't actually contribute to the view's width